cpp/modules/luma.gl/garrow/src/array.cc

```diff
@@ -44,36 +44,32 @@ void Array::setData(const std::shared_ptr<arrow::Array>& data, wgpu::BufferUsage
   auto vertexSize = getVertexFormatSize(vertexFormat);
 
   if (!this->_buffer || data->length() != this->_length) {
-    auto size = vertexSize * data->length();
-    this->_buffer = this->_createBuffer(this->_device, size, usage);
+    auto bufferSize = vertexSize * data->length();
+    this->_buffer = this->_createBuffer(this->_device, bufferSize, usage);
   }
 
   // TODO(ilija@unfolded.ai): Handle arrays with null values correctly
   if (data->null_count() > 0) {
     throw new std::runtime_error("Data with null values is currently not supported");
   }
 
-  // TODO(ilija@unfolded.ai): Not sure if this is the best way to extract the data out of any data type
+  // If child_data isn't empty, data is a list array
+  if (!data->data()->child_data.empty()) {
+    uint64_t offset = 0;
 
-  uint64_t offset = 0;
-  auto buffers = data->data()->buffers;
-  // Starting from buffer at index 1, as the first buffer is used for null bitmap
-  for (int i = 1; i < buffers.size(); i++) {
-    auto buffer = buffers[i];
+    // Go through child data for list data types and use child buffers
```

Comment:
I got lost when I read this comment. I can understand that we are iterating through "chunks" of a data array, but that is different from "children" in my understanding. 

Children is related to the hierarchy of column types, (a list having a child being the primitive element of the type, a struct having multiple children etc).

Be extra careful distinguishing between chunks and children, and maybe make an short "ASCII diagram" in comments above explain what children contains in the list case?

Reply:
There are no chunks in this use case. There's a separate type called `ChunkedArray` that a table contains, we're working with non-chunked `Array`s here which are result of accessor processing.

We're dealing with children here, which as you noted, covers the case of list arrays. We iterate over each of the arrays in the list and extract its data

Reply:
I think what gets me here is that my mental image is that a list array only has one child, so this iteration is confusing - child is a very generic term and I wish arrow had used a more specific word.